### DISCUSSION TITLE
client_id is required when defining a custom provider

### DISCUSSION THREAD

**Discussion by yalegria** [nextauthjs/next-auth#8081] [Help] [answered]
Using nextjs 13.2.3 and next-auth 4.22.1. 

I setup a custom provider object to authenticate against a Oauth provider. After clicking on the provisioned button "Sign in with Custom Provider" I got a error message "Try signing with a different account".

On the logs, I see an error as welll:

```
ile-ml/node_modules/next/dist/server/base-server.js:157:99',
    name: 'TypeError'
  },
  providerId: 'rutgers-cas',
  message: 'client_id is required'
}
```

Based on the docs, the customer provider object does not have a 'client_id' property.

**Comment by yalegria** [comment, marked as answer]
Apparently property client_id is required in the object defining a custom provider. However, the correct label for the property is clientId. I added that to the object and the error is gone.

**Comment by reginpv** [comment]
Same issue with my app, I just noticed switching between branches remove my env.local (which I manage via Vercel).

If you have the same setup just pull your env `vercel env pull` then run dev again.

Overall the problem looks like it can not find the `client_id` make sure it is properly added on your env.

**Comment by Yudeat** [comment]
i have tried all the method you people mention above but none of them are ready to work so which others solution can be used?
